### PR TITLE
Add a base structure for adoption module

### PR DESCRIPTION
**Pull request by belgamo**
Continuing #121 

### Summary
This PR is to add the core module of our app: adoptions. This is actually a POC for our v2 and I want to implement something on the mobile side integrating it with our backend to see how it goes.

### Validation
On NestJS we have a couple of validation strategies: [schema-based](https://docs.nestjs.com/pipes#schema-based-validation) and [decorator-based](https://docs.nestjs.com/pipes#class-validator). I decided to go with the schema-based one because then we can share the schema across our apps. That's I created the `schema-validations` under the shared folder. Also, I'm using [Joi](https://github.com/hapijs/joi) to validate. We have a custom pipe on the backend that integrates Nest with the Joi schemas.

### `schema-validations` packages
So as I said above I added this new package to share schemas. I had to set up typescript properly to generate a dist folder with the transpiled code. Due to this, whenever we make a change on `schema-validations` we need to rebuild this package so the backend app can have access to the new code. To rebuild, we just run: `pnpm -r run build`. Not sure if we can automate this task by watching the files in the workspace. Maybe `Nx` can help with this, but let's do that in the future.

### CODE REVIEW

File: shared/validation-schemas/src/adoption.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  type: Joi.string().max(5),
+
+  age: Joi.number(),
+
+  size: Joi.string().max(6),
+
+  observations: Joi.string().allow("").max(100),
+
+  photos: Joi.array().optional().items(Joi.string()),
```
Comment (abarbosa-godaddy) [inline shared/validation-schemas/src/adoption.ts]: These empty lines. Should we get rid of them?

**Comment by abarbosa-godaddy** [comment]
Worth it to add unit tests to validators and controllers at least?

File: shared/validation-schemas/package.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "@animavita/validation-schemas",
```
Comment (brunohkbx) [inline shared/validation-schemas/package.json:2]: Why this need to be shared?

File: shared/validation-schemas/src/adoption.ts (hunk 1 above)
Comment (brunohkbx) [inline shared/validation-schemas/src/adoption.ts]: Is it possible to use `Joi.string().uri()` instead?

Comment (belgamo) [reply shared/validation-schemas/src/adoption.ts]: I think so, let's try.

File: shared/validation-schemas/package.json (hunk 2 above)
Comment (belgamo) [reply shared/validation-schemas/package.json:2]: Cause then we can validate on the client as well matching the same schema.

File: shared/validation-schemas/src/adoption.ts (hunk 1 above)
Comment (belgamo) [reply shared/validation-schemas/src/adoption.ts]: I like this style cause we might have to nest more Joi functions.

**Comment by belgamo** [comment]
> Worth it to add unit tests to validators and controllers at least?

Done!

**Review by gquaresma-godaddy — APPROVED** [review]

**Pull request merged**